zkevm-circuits/src/pi_circuit2.rs

```diff
@@ -91,6 +148,39 @@ pub struct ExtraValues {
     prev_state_root: H256,
 }
 
+#[derive(Debug, Clone)]
+struct BlockhashColumns {
+    blk_hdr_rlp: Column<Advice>,
+    blk_hdr_rlp_inv: Column<Advice>,
+    blk_hdr_rlp_const: Column<Fixed>,
+    q_blk_hdr_rlp: Selector,
+    q_blk_hdr_rlp_const: Selector,
+    blk_hdr_rlp_len_calc: Column<Advice>,
+    blk_hdr_rlp_len_calc_inv: Column<Advice>,
+    q_blk_hdr_total_len: Selector,
+    blk_hdr_reconstruct_value: Column<Advice>,
+    q_parent_hash: Column<Fixed>,
+    q_beneficiary: Column<Fixed>,
+    q_state_root: Column<Fixed>,
+    q_transactions_root: Column<Fixed>,
+    q_receipts_root: Column<Fixed>,
+    q_number: Column<Fixed>,
+    // TODO(George) gas limit is u64 and not u256
+    q_gas_limit: Column<Fixed>,
+    q_gas_used: Column<Fixed>,
+    q_timestamp: Column<Fixed>,
+    q_mix_hash: Column<Fixed>,
+    q_base_fee_per_gas: Column<Fixed>,
+    q_withdrawals_root: Column<Fixed>,
+    q_hi: Selector,
+    q_lo: Column<Fixed>,
```

Comment:
It seems that these fixed columns could be replaced by maybe 1-2.

They are meant for distinguishing the different header fields but 
if we change the reconstructing of values to RLC (see comment https://github.com/taikoxyz/zkevm-circuits/pull/98/files?show-viewed-files=true&file-filters%5B%5D=#r1230694048) then there is no need to distinguish them really.

Reply:
optimisation implemented. Net gain of 8 fixed columns and 10 lookup operations.